Reorganize docs contrib guide

mistyhacks opened the pull request:
Preview at https://deploy-preview-9510--kubernetes-io-master-staging.netlify.com/docs/contribute/

This is a HUGE PR to review. I'm very sorry about that. It does the following major things:

- Rewrite, reorganize, and consolidate all docs about contributing under a new Contribute section of the website, under `/docs/contribute/`
- Improve and update docs about page templates, content organization, etc
- Change the behavior of the "Edit this page" links (at bottom and top right) to go straight to the Github editor with no preliminary page in between
- Change the text of the "Create an issue" button to "Report a problem". Include both the page URL and file name in the PR subject. I'd love to have left the subject blank to force the reporter to write a meaningful title, and include the path and file name in the issue body, but that overrides the Github issue template. Also remove reliance on Javascript to build the "Create an issue" link.
- Lots of redirects added

There are still some whitespace errors (trailing, mostly) that I need to find and fix. But this is ready to review.

mistyhacks commented:
/hold cancel

mistyhacks commented:
/assign @tallt0m 
/assign @chenopis 
/assign @jaredbhatti 
/assign @zacharysarah 
/assign @Bradamant3

tengqm commented on content/en/docs/contribute/style/write-new-topic.md:
```diff
@@ -70,43 +71,47 @@ triple-dashed lines at the top of the page. Here's an example:
+use multiples of 10, to account for adding topics later. For instance, a topic
+with weight `10` will come before one with weight `20`.
 
 ## Including code from another file
```
I have a PR under review for this sub-topic. #9574

mistyhacks replied:
@tengqm Can you let me know when your PR is merged so I can make sure those changes get into this branch? I'm sure your PR will be ready before mine has been completely reviewed.

mistyhacks replied:
I just merged it. Let me now rebase this PR and integrate that content into the new location.

bradtopol commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,4 +1,5 @@
 <!-- Thanks for filing an issue! Before submitting, please fill in the following information. -->
+<!-- See https://kubernetes.io/docs/contribute/start/ for guidance on writing an actionable issue description. -->
```
Nice!  Very cool to have a helpful pointer here.

bradtopol commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,6 +1,7 @@
 >^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+> Please delete this note before submitting the pull request.
 > For 1.12 Features: set Milestone to 1.12 and Base Branch to release-1.12
```
Do we have instructions somewhere that provides an example on how to set the base branch to release-1.12?  For folks new to git that would be very helpful.  Also do we have instructions somewhere on setting the milestone. People will ask how to do that as well

bradtopol commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,6 +1,7 @@
 >^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+> Please delete this note before submitting the pull request.
 > For 1.12 Features: set Milestone to 1.12 and Base Branch to release-1.12
+> For help editing and submitting pull requests, see https://kubernetes.io/contribute/start/
```
Very helpful!

bradtopol commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,57 @@
+and whether you self-identify as a developer, an end user, or someone who just
+can't stand seeing typos.
+
+Looking for the [style guide](/docs/contribute/style/style-guide/)?
```
Instead of "Looking for the"   Perhaps say, Before you start contributing we highly recommend you review our [style guide] .  This will help to get your pull requests merged more quickly.

bradtopol commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,57 @@
+  - Participate in a Kubernetes release team as a docs representative
+  - Propose improvements to the style guide
+  - Propose improvements to docs tests
+  - Propose improvements to the Kubernetes website or other tooling
```
Should Approve and Merge PRs be on this list?

bradtopol commented on content/en/docs/contribute/advanced.md:
```diff
@@ -0,0 +1,70 @@
+currently work and why past decisions have been made before proposing sweeping
+changes. The quickest way to get answers to questions about how the documentation
+currently works is to ask in the #sig-docs Slack channel on
+[kubernetes.slack.com](https://kubernetes.slack.com)
```
Great section!

bradtopol commented on content/en/docs/contribute/participating.md:
```diff
@@ -0,0 +1,105 @@
+
+### Approvers
+
+Approvers have the ability to merge a PR.
```
I thought this was called maintainer

mistyhacks replied:
Updated.

bradtopol commented on content/en/docs/contribute/style/page-templates.md:
```diff
@@ -0,0 +1,235 @@
+
+{{% capture overview %}}
+
+When contributing new topics, apply one of the following templates to them.
```
So awesome to have this documented!!!!

bradtopol commented on content/en/docs/contribute/style/page-templates.md:
```diff
@@ -0,0 +1,235 @@
+    | discussion    | no        |
+    | whatsnext     | no        |
+
+    The page's body will look like this (without any optional sections you don't
```
So are any of the sections below optional or are they mandatory? Its unclear to me

mistyhacks replied:
I don't understand. The "Required" column in the table should cover this, I thought.

mistyhacks commented on content/en/docs/contribute/style/page-templates.md:
```diff
@@ -0,0 +1,235 @@
+`/content/en/docs/tutorials` directory, with the following characteristics:
+
+- In the page's YAML front-matter, set `content_template: templates/tutorial`.
+- In the page's body, set the following `capture` variables:
```
I think the inline examples show it pretty well, right?

mistyhacks commented on content/en/docs/contribute/style/page-templates.md:
```diff
@@ -0,0 +1,235 @@
+  - For `overview`, use a paragraph to set context for the entire topic.
+  - Add the `{{< toc >}}` shortcode to show an in-page table of contents.
+  - For `prerequisites`, use bullet lists when possible. Add additional
+    prerequisites below the ones included by default.
```
I added some clarification.

mistyhacks commented:
Addressed all comments so far.

mistyhacks commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,57 @@
+
+This list is divided into things anyone can do, things Kubernetes project
+members can do, and things Kubernetes maintainers can do. The intent is not to
+throttle contributions or limit who can do what. Some of the more advanced tasks
```
Do we have guidelines on expected time commitment?

chenopis replied:
We don't have guidelines on expected time commitment; however, for active Maintainers, IMHO we should set the expectation that they be part of the PR wrangler rotation, which is maybe 5-10 hours for the week they are on rotation. Also, they should regularly attend the SIG Docs meeting, but it's hard to say what "regularly" means. Maybe at least monthly?

chenopis commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,57 @@
+---
+content_template: templates/concept
```
As an aside, maybe we should have a generic template because we seem to be using the concept template one for anything that is not a Task or Tutorial.

mistyhacks commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,57 @@
+This is not an exhaustive list of ways you can contribute to the Kubernetes
+documentation, but it should get your juices flowing.
+
+This list is divided into things anyone can do, things Kubernetes project
```
The Github group is called sig-docs-maintainers. I think that's confusing.

mistyhacks commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,57 @@
+and organizational decisions that have already been made.
+
+- [Anyone](start/)
+  - File actionable bugs
```
I feel like that is less clear. As a new contributor, it seems more clear to me what a "bug" is than an "issue" and "File" seems more active than "open". Thoughts?

zacharysarah replied:
We’re describing GitHub functionality, so I think it makes more sense to describe what users see rather than what we assume they’re familiar with.

mistyhacks commented:
Addressed new feedback, some questions are outstanding and may be obscured by the Github UI because those lines of the source have changed. @zacharysarah those questions are for you primarily.

mistyhacks commented:
/assign @cody-clark

mistyhacks commented:
Also I will wait for a little more consensus on time commitment and specific benchmarks for being added to SIG Docs, becoming a reviewer, and becoming an approver.

mistyhacks commented:
/cc @cody-clark

kbhawkey commented on content/en/docs/contribute/_index.md:
```diff
@@ -0,0 +1,73 @@
+
+{{% capture overview %}}
+
+If you would like to help contribute to the Kubernetes documentation or website,
```
A small nit:
The left nav item for `start contributing` is just `start`. How about 'Getting Started', or a different way to say, beginner?

mistyhacks replied:
It's under a Contribute heading so I thought it would be fine to be concise. I wanted a strong call-to-action.

mistyhacks replied:
Update to say I addressed this by changing it to "Start contributing".

kbhawkey commented on content/en/docs/contribute/start.md:
```diff
@@ -0,0 +1,356 @@
+## Write a blog post
+
+Anyone can write a blog post and submit it for review. Blog posts should not be
+commercial in nature and should consiste of content that will apply broadly to
```
fix: consiste

kbhawkey commented on content/en/docs/contribute/style/page-templates.md:
```diff
@@ -0,0 +1,239 @@
+    
+    {{% capture prerequisites %}}
+    
+    {{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
```
Why is this basic version check necessary for every task/tutorial? I could understand using a version check if the instructions in the task pertain to a specific version, otherwise, it seems odd to remind the user to check their version without saying why.

mistyhacks replied:
This was decided by the SIG when the templates were decided upon. Not my new change. :)

zacharysarah commented:
From [Slack](https://kubernetes.slack.com/archives/C1J0BPD2M/p1532983678000478) with @mistyhacks: 
> still awaiting some clarification on org member / sig review / sig approver, what the specific time requirements and time-commit expectations are of each, for the docs contributor guid

Here's a summary of the operative criteria. Hopefully I got everything--other maintainers, please chime in!

Role | Docs prerequisites | Responsibilities
---|---|---
[Member](https://github.com/kubernetes/community/blob/master/community-membership.md#member) | - PRs must be to k/website | Follow the [code of conduct](https://kubernetes.io/community/code-of-conduct/)
[Reviewer](https://github.com/kubernetes/community/blob/master/community-membership.md#reviewer) | - As per member<br>- Regularly participate in weekly SIG meetings  | - As per member<br>- Review PRs auto-assigned by blunderbuss within 1 week
[Approver](https://github.com/kubernetes/community/blob/master/community-membership.md#approver) | - As per reviewer<br>- Contribute regularly for at least 3 months<br>- Knowledgeable with SIG Docs workflows and tooling | - As per reviewer<br>- Respond to PR comments/requests for review from other maintainers within 1 week<br>- Serve periodically as the [PR Wrangler](https://github.com/kubernetes/website/wiki/PR-Wranglers) for SIG Docs

kbhawkey commented on content/en/docs/contribute/start.md:
```diff
@@ -0,0 +1,356 @@
+
+When you are comfortable with all of the tasks discussed in this topic and you
+want to engage with the Kubernetes docs team in deeper ways, read the
+[internediate docs contribution guide](/docs/contribute/intermediate/).
```
spelling nit: internediate

kbhawkey commented on content/en/docs/contribute/start.md:
```diff
@@ -0,0 +1,356 @@
+    docs.
+
+2.  By default, the only filter that is applied is `open`, so you don't see
+    pull reuests that have already been closed or merged. It's a good idea to
```
spelling nit: reuests

mistyhacks commented:
I rebased, fixed the nits pointed out by @kbhawkey , and updated the `participating.md` with more details and to include SIG Docs chair. Thanks everyone for your reviews. I think barring anything major that is wrong here, this is ready to go in and we can iterate.

/assign @chenopis

mistyhacks commented:
Troubleshooting netlify failure.

chenopis commented:
Netlify build looks good:
> 3:39:04 PM: Build ready to start
3:39:05 PM: build-image version: 7c49b266ed8abd833dee6036ec0a4b9ee887658a
3:39:05 PM: buildbot version: f58df58ab0bceffdcb8d1e60fec992495a2a6751
3:39:05 PM: Fetching cached dependencies
3:39:06 PM: Starting to download cache of 314.6MB
3:39:07 PM: Finished downloading cache in 1.403494345s
3:39:07 PM: Starting to extract cache
3:39:13 PM: Finished extracting cache in 6.232225962s
3:39:13 PM: Finished fetching cache in 7.787273586s
3:39:13 PM: Starting to prepare the repo for build
3:39:14 PM: Preparing Git Reference pull/9510/head
3:39:20 PM: Found netlify.toml. Overriding site configuration
3:39:20 PM: Different build command detected, going to use the one specified in the toml file: 'hugo --enableGitInfo -b $DEPLOY_PRIME_URL' versus 'hugo' in the site
3:39:20 PM: Starting build script
3:39:20 PM: Installing dependencies
3:39:21 PM: Started restoring cached node version
3:39:23 PM: Finished restoring cached node version
3:39:24 PM: v8.11.3 is already installed.
3:39:25 PM: Now using node v8.11.3 (npm v5.6.0)
3:39:25 PM: Attempting ruby version 2.2.9, read from environment
3:39:26 PM: Using ruby version 2.2.9
3:39:26 PM: Using PHP version 5.6
3:39:26 PM: Installing Hugo 0.46
3:39:26 PM: Started restoring cached go cache
3:39:26 PM: Finished restoring cached go cache
3:39:26 PM: unset GOOS;
3:39:27 PM: unset GOARCH;
3:39:27 PM: export GOROOT='/opt/buildhome/.gimme/versions/go1.10.linux.amd64';
3:39:27 PM: export PATH="/opt/buildhome/.gimme/versions/go1.10.linux.amd64/bin:${PATH}";
3:39:27 PM: go version >&2;
3:39:27 PM: export GIMME_ENV='/opt/buildhome/.gimme/env/go1.10.linux.amd64.env';
3:39:27 PM: go version go1.10 linux/amd64
3:39:27 PM: Installing missing commands
3:39:27 PM: Verify run directory
3:39:27 PM: Executing user command: hugo --enableGitInfo -b $DEPLOY_PRIME_URL
3:39:29 PM: Building sites …
3:39:37 PM: 

> 3:39:37 PM:                    |  EN  | CN
3:39:37 PM: +------------------+------+-----+
3:39:37 PM:   Pages            | 1339 | 163
3:39:37 PM:   Paginator pages  |  232 |   0
3:39:37 PM:   Non-page files   |  405 | 267
3:39:37 PM:   Static files     |  803 | 803
3:39:37 PM:   Processed images |    0 |   0
3:39:37 PM:   Aliases          |    3 |   1
3:39:37 PM:   Sitemaps         |    2 |   1
3:39:37 PM:   Cleaned          |    0 |   0
3:39:37 PM: Total in 8030 ms
3:39:37 PM: Caching artifacts
3:39:37 PM: Started saving pip cache
3:39:37 PM: Finished saving pip cache
3:39:37 PM: Started saving emacs cask dependencies
3:39:37 PM: Finished saving emacs cask dependencies
3:39:37 PM: Started saving maven dependencies
3:39:37 PM: Finished saving maven dependencies
3:39:37 PM: Started saving boot dependencies
3:39:37 PM: Finished saving boot dependencies
3:39:37 PM: Started saving go dependencies
3:39:37 PM: Finished saving go dependencies
3:39:37 PM: Build script success
3:39:37 PM: Starting to deploy site from 'public'
3:40:32 PM: Starting post processing
3:41:07 PM: Finished processing build request in 2m1.88368775s
3:43:02 PM: Post processing done
3:43:02 PM: Site is live

mistyhacks commented:
Confirmed the last commit fixed the bug that was breaking Netlify and also adds a related test to the smoke tests page. This is ready to go. 🍾

chenopis commented:
/lgtm
/approve
